File: ESTRUCURAS_DE_DATOS/Cola/typeScriptEjercicios/numeros.ts
class Queue {
    private items: number[] = [];

    isEmpty(): boolean{
        return this.items.length === 0;
    }

    enqueue (item: number): void{

        this.items.push(item);

    }

    dequeue(): number | undefined {
        return this.items.shift();
    }

    front (): number {

        return this.items[0];
    }

    rear (): number{
        return this.items[this.size() - 1];
    }

    size (): number {
        return this.items.length;
    }

    
}

const queue = new Queue();

queue.enqueue(10);
queue.enqueue(20);
queue.enqueue(30);
queue.enqueue(40);

console.log('Size queue: ' , queue.size());
console.log('first queue: ' , queue.front());
console.log('Last queue: ' , queue.rear());

console.log('Elementos eliminados: ');
while (!queue.isEmpty) {
    console.log(queue.dequeue());
}

console.log('Size queue: ' , queue.size());





    

File: ESTRUCURAS_DE_DATOS/Cola/pythonEjercicios/numeros.py
class Queue:

    def __init__(self):
        self.number = []

    def is_empty(self):
        return len(self.number) == 0 

    def enqueue (self, number):
        self.number.append(number)

    def dequeue(self):
        if not self.isEmpty:
            return self.number.pop()

    def front (self):
        return self.number[0]

    def rear (self):
        if not self.isEmpty:
            return self.number[-1]

    def size (self):
        print(len(self.number))

    
numbers = Queue()

numbers.enqueue(10)
numbers.enqueue(20)
numbers.enqueue(30)
numbers.enqueue(40)

print("Size queue: " , numbers.size)
print("First queue: " , numbers.front)
print("First queue: " , numbers.rear)

print("Elementos eliminados: ")

while (numbers.is_empty != 0):
    print(numbers.dequeue)

print("Size queue: " , numbers.size)









    
    
